src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -266,14 +280,33 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // - RoutedEventArgs: not used
     // Return Value:
     // - <none>
-    void TermControl::_CloseSearchBoxControl(const winrt::Windows::Foundation::IInspectable& /*sender*/, RoutedEventArgs const& /*args*/)
+    void TermControl::_CloseSearchBoxControl(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                             RoutedEventArgs const& /*args*/)
     {
         _searchBox->Visibility(Visibility::Collapsed);
 
         // Set focus back to terminal control
         this->Focus(FocusState::Programmatic);
     }
 
+    // Method Description:
+    // - This event handler works to give the search box an attempt to process
+    //   keybindings. Most notably, we need this for the findPrevious and
+    //   findNext actions to be actionable from within the search box.
+    void TermControl::_SearchKeyHandler(Windows::Foundation::IInspectable const& /*sender*/,
+                                        Input::KeyRoutedEventArgs const& e)
+    {
+        const auto modifiers = _GetPressedModifierKeys();
+        const auto vkey = gsl::narrow_cast<WORD>(e.OriginalKey());
+        const auto scanCode = gsl::narrow_cast<WORD>(e.KeyStatus().ScanCode);
+
+        if (!modifiers.IsAltGrPressed() &&
+            _TryHandleKeyBinding(vkey, scanCode, modifiers))
+        {
+            e.Handled(true);
+        }
+    }

```

Comment:
this is the bit that scares me the most. Right now, there are folks with bindings that do things ot the terminal (like rebinding ctrl+arrow keys, etc.) and they will not be expecting them to bypass the search box. I _understand_ why we have to do this, and I would hate to grow a bunch of special cases all over the code (literally everywhere!) that detect certain key bindings and act differently for them, but.. ugh.

Now, here's a question. Are problems like this solved by having actions report which modes they work in? We've talked about modal actions before, but.. this is different. Actions that must be tested in search, actions that must be tested in settings, and actions that must be tested in terminals are different but we have a unified interface for them.

Reply:
I don't disagree that this is weird. How the search box is tied in with the rest of the control is already weird, and this doesn't make it less weird. I'm almost of the mind that it should be something that lives up in TerminalApp, and like, has events that trigger callbacks into the control. Then again, I suppose even embedded in another app, the control should have a find dialog. \</showerthought>

I really don't think we should have the KeyMapping down in the control layer, so we probably do need a better way of handling this. 

Should I just yank this part of the PR while we sort out this thought experiment? I'm not particularly attached to it currently.

Reply:
@DHowett Turns out, I didn't need to do this at all. By moving the search box check lower (in my latest commit(s)), we can get search box handling of keybindings for free. So no extra check here. The new commit seems much, much more reasonable.